src/probnum/prob/distributions/distribution.py

```diff
@@ -72,21 +74,53 @@ def __init__(self, parameters=None, pdf=None, logpdf=None, cdf=None, logcdf=None
         self._sample = sample
         self._mean = mean
         self._cov = cov
+        self._set_shape(shape)
         self._dtype = dtype
         self._random_state = scipy._lib._util.check_random_state(random_state)
 
     def _check_distparams(self):
         pass
         # TODO: type checking of self._parameters; check method signatures.
 
+    def _set_shape(self, shape):
+        """
+        Sets shape in accordance with distribution mean.
+        """
+        self._shape = shape
+        try:
+            # Set shape based on mean
+            if np.isscalar(self.mean()):
+                shape_mean = ()
+            else:
+                shape_mean = self.mean().shape
+            if shape is None or shape_mean == shape:
+                self._shape = shape_mean
+            else:
+                raise ValueError("Shape of distribution mean and given shape do not match.")
```

Comment:
whenever I see code blocks with a leading comment, I personally think it would be more readable to extract this code into a function, i.e. `self. _set_shape_based_on_mean(...)` and so on which makes each function nice and short (and the code self-documenting). I understand that this is just personal preference.